evm/src/cpu/kernel/interpreter.rs

```diff
@@ -306,13 +365,40 @@ fn find_jumpdests(code: &[u8]) -> Vec<usize> {
 
 #[cfg(test)]
 mod tests {
-    use crate::cpu::kernel::interpreter::run;
+    use hex_literal::hex;
+
+    use crate::cpu::kernel::interpreter::{run, Interpreter};
 
     #[test]
     fn test_run() {
         let code = vec![
             0x60, 0x1, 0x60, 0x2, 0x1, 0x63, 0xde, 0xad, 0xbe, 0xef, 0x56,
         ]; // PUSH1, 1, PUSH1, 2, ADD, PUSH4 deadbeef, JUMP
-        assert_eq!(run(&code, 0, vec![]), vec![0x3.into()]);
+        assert_eq!(run(&code, 0, vec![]).stack, vec![0x3.into()]);
+    }
+
+    #[test]
+    fn test_run_with_memory() {
+        //         PUSH1 0xff
+        //         PUSH1 0
+        //         MSTORE
+
+        //         PUSH1 0
+        //         MLOAD
+
+        //         PUSH1 1
+        //         MLOAD
+
+        //         PUSH1 0x42
+        //         PUSH1 0x27
+        //         MSTORE8
+        let code = vec![
+            0x60, 0xff, 0x60, 0x0, 0x52, 0x60, 0, 0x51, 0x60, 0x1, 0x51, 0x60, 0x42, 0x60, 0x27,
+            0x53,
+        ];
+        let run = run(&code, 0, vec![]);
+        let Interpreter { stack, memory, .. } = run;
+        assert_eq!(stack, vec![0xff.into(), 0xff00.into()]);
+        assert_eq!(&memory.memory, &hex!("00000000000000000000000000000000000000000000000000000000000000ff0000000000000042000000000000000000000000000000000000000000000000"));
```

Comment:
It seems like the 42 is in `memory[39]` in the hex, but shouldn't it be at `memory[27]`? I'm probably missing something since the code looks right

Reply:
It should be at location 0x27 = 39.

Reply:
Ah of course :)